### PR TITLE
Fixed the navX calibration bug hopefully!

### PR DESCRIPTION
**Pull request by imathno**

### CODE REVIEW

File: src/xbot/common/subsystems/pose/BasePoseSubsystem.java [hunk 1]
```diff
@@ -40,18 +40,20 @@
...
         log.info("Creating");
+        this.classInstantiationTime = Timer.getFPGATimestamp();
         imu = factory.getGyro(ImuType.navX);
         
         currentHeadingProp = propManager.createEphemeralProperty("CurrentHeading", 0.0);
         // Right when the system is initialized, we need to have the old value be
         // the same as the current value, to avoid any sudden changes later
-        Timer.delay(1);
         currentHeading = new ContiguousHeading(0);
-        setCurrentHeading(FACING_AWAY_FROM_DRIVERS);
```
Comment (aschokking) [inline src/xbot/common/subsystems/pose/BasePoseSubsystem.java]: What will currentHeading return to callers before the navx is ready?

File: tests/xbot/common/subsystems/pose/BasePoseTest.java [hunk 2]
```diff
@@ -17,9 +18,11 @@
 public class BasePoseTest extends BaseWPITest {
 
     protected TestPoseSubsystem pose;
+    private MockTimer mockTimer;
```
Comment (aschokking) [inline tests/xbot/common/subsystems/pose/BasePoseTest.java]: planning to add a test?

File: src/xbot/common/subsystems/pose/BasePoseSubsystem.java (hunk 1 above)
Comment (imathno) [inline src/xbot/common/subsystems/pose/BasePoseSubsystem.java]: 0

Comment (JohnGilb) [inline src/xbot/common/subsystems/pose/BasePoseSubsystem.java]: Document that in a javadoc comment on the getCurrentHeading method

File: src/xbot/common/subsystems/pose/BasePoseSubsystem.java [hunk 3]
```diff
@@ -186,8 +187,19 @@ public void getYawAngularVelocity(){
...
+        return isNavXReady;
+    }
+    
     @Override
     public void updatePeriodicData() {
-        updatePose();
-    }
-}
\ No newline at end of file
+        log.info("test");
```
Comment (JohnGilb) [inline src/xbot/common/subsystems/pose/BasePoseSubsystem.java]: remove this log line

File: src/xbot/common/subsystems/pose/BasePoseSubsystem.java [hunk 4]
```diff
@@ -186,8 +187,19 @@ public void getYawAngularVelocity(){
...
+        if (!isNavXReady && (classInstantiationTime + 1 < Timer.getFPGATimestamp())) {
+            System.out.println("REACHED");
```
Comment (JohnGilb) [inline src/xbot/common/subsystems/pose/BasePoseSubsystem.java]: make this say something useful, like "Performing initial gyro calibration"

File: tests/xbot/common/subsystems/pose/PoseSubsystemTest.java [hunk 5]
```diff
@@ -13,6 +13,7 @@
     
     @Before
```
Comment (JohnGilb) [inline tests/xbot/common/subsystems/pose/PoseSubsystemTest.java]: You don't actually add a test that tests your new behavior.

File: src/xbot/common/subsystems/pose/BasePoseSubsystem.java (hunk 1 above)
Comment (imathno) [inline src/xbot/common/subsystems/pose/BasePoseSubsystem.java]: I added it

**Pull request merged**